### PR TITLE
Add default testing framework and unit test to aws-nodejs

### LINKED ISSUE #3436: Default tests in aws-nodejs template
# This is a Feature Proposal

I propose we add a testing framework and a single unit test by default.

## Description

For bug reports:
* What went wrong?
* What did you expect should have happened?
* What was the config you used?
* What stacktrace or error message from your provider did you see?

For feature proposals:
* What is the use case that should be solved. The more detail you describe this in the easier it is to understand for us.
* If there is additional config how would it look

Right now, the `aws-nodejs` template does not come with any tests. Many frameworks come with a default test configuration that run a simple unit test with `npm test` to make it easier for newbies to start writing tests. 

This would require the addition of something like `jest` and `babel-jest` to `package.json` and a single file that runs a unit test on the `hello` function.

I'm happy to put together a PR for this since I've had to add a testing framework every time I generate a new Serverless project. Let me know if there's any interest.

Similar or dependent issues:
* #12345

## Additional Data

* ***Serverless Framework Version you're using***:
* ***Operating System***:
* ***Stack Trace***:
* ***Provider Error messages***:

### PR DESCRIPTION
**Pull request by samcorcos**
## What did you implement:

Closes #3436

## How did you implement it:

I added `jest`, the packages that jest requires, and a single unit test that simply tests the `hello` function to ensure it was called with the proper data.

## How can we verify it:

Generate a new project with `aws-nodejs`, run `npm install && npm test`. You should get a passing test as well as an obvious starting point for writing more tests.

## Todos:

- [ ] Write tests
- [ ] Write documentation
- [ ] Fix linting errors
- [ ] Make sure code coverage hasn't dropped
- [ ] Provide verification config / commands / resources
- [ ] Enable "Allow edits from maintainers" for this PR
- [ ] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

**Comment by pmuens** [comment]
Thanks for the PR @samcorcos 👍 

Cross-linking to the relevant issue here (#3436) again where more discussion can be found.

My thoughts on this is that I'm not sure if we should add more boilerplate code to the core template. I see the value because once you get more professional you need to have some testing framework in place.

However choosing the right framework (I'm also a big fan of Jest) will be always opinionated. Furthermore it might be overwhelming for users who just want to start out and deploy their first serverless function.

Maybe we can add a more complex template to the [examples](https://github.com/serverless/examples) repository where users can see how testing Lambdas works.

File: lib/plugins/create/templates/aws-nodejs/package.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  "name": "aws-nodejs",
+  "version": "1.0.0",
+  "description": "",
+  "main": "handler.js",
+  "scripts": {
+    "test": "jest"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
```
Comment (vladgolubev) [inline lib/plugins/create/templates/aws-nodejs/package.json]: Maybe `MIT`?

File: lib/plugins/create/templates/aws-nodejs/package.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  "devDependencies": {
+    "babel-jest": "^19.0.0",
+    "babel-preset-es2015": "^6.24.0",
+    "babel-preset-stage-0": "^6.22.0",
```
Comment (vladgolubev) [inline lib/plugins/create/templates/aws-nodejs/package.json]: https://github.com/babel/babel-preset-env is now a recommended official approach

**Comment by pmuens** [comment]
@samcorcos thanks again for this addition.

I'll close this one for now since the conclusion is that the main core templates should be as simple and not too opinionated as possible. Would be nice to have this in the `examples` repository like we've discussed in #3436.

Thanks again for your PR and the contribution! 👍

**Pull request closed without merging**